internal/api/oidc/token_exchange_converter.go

```diff
@@ -0,0 +1,99 @@
+package oidc
+
+import (
+	"time"
+
+	"github.com/zitadel/oidc/v3/pkg/oidc"
+
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/query"
+)
+
+type exchangeToken struct {
+	tokenType     oidc.TokenType
+	userID        string
+	issuer        string
+	resourceOwner string
+	authTime      time.Time
+	authMethods   []domain.UserAuthMethodType
+	actor         *domain.TokenActor
+	audience      []string
+	scopes        []string
+}
+
+func (et *exchangeToken) nestedActor() *domain.TokenActor {
+	return &domain.TokenActor{
+		Actor:  et.actor,
+		UserID: et.userID,
+		Issuer: et.issuer,
+	}
+}
+
+func accessToExchangeToken(token *accessToken, issuer string) *exchangeToken {
+	return &exchangeToken{
+		tokenType:     oidc.AccessTokenType,
+		userID:        token.userID,
+		issuer:        issuer,
+		resourceOwner: token.resourceOwner,
+		authMethods:   token.authMethods,
+		actor:         token.actor,
+		audience:      token.audience,
+		scopes:        token.scope,
+	}
+}
+
+func idTokenClaimsToExchangeToken(claims *oidc.IDTokenClaims) *exchangeToken {
+	resourceOwner, _ := claims.Claims[ClaimResourceOwnerID].(string)
```

Comment:
if the claim is missing the exchange will not work and return an error
```json
{
	"error": "invalid_request",
	"error_description": "context missing"
}
```
i think the integration test works, since you use a PAT for the initial exchange where the returned id_token contains such claim